src/plots/xyPlot.ts

```diff
@@ -135,15 +237,28 @@ export class XYPlot<X, Y> extends Plot {
     var adjustCallback = key === XYPlot._X_KEY ? this._adjustYDomainOnChangeFromXCallback
                                                : this._adjustXDomainOnChangeFromYCallback;
     scale.onUpdate(adjustCallback);
+
+    if (this._lazyDomainChange) {
+      scale.offUpdate(this._renderCallback);
```

Comment:
Could we have a separate function that switches between them only for the X and Y `Scale`s?
```Typescript
this._xScaleRenderCallback = (scale) => {
  if (this.lazyDomainChange()) {
    this._lazyDomainChangeCallbackX(scale);
  } else {
    this._renderCallback(scale);
  }
};

...

protected _installScaleForKey(scale: Scale<any, any>, key: string) {
  if (key === XYPlot._X_KEY) {
    scale.onUpdate(this._xScaleRenderCallback);
  } else {
    super._installScaleForKey(scale, key);
  }
}
```

Not sure this is better, though.